web_services/foxx/brightid/db.js

```diff
@@ -319,168 +166,145 @@ function invite(inviter, invitee, groupId, data, timestamp) {
 }
 
 function dismiss(dismisser, dismissee, groupId, timestamp) {
-  if (! groupsColl.exists(groupId)) {
-    throw new errors.GroupNotFoundError(groupId);
-  }
-  const group = groupsColl.document(groupId);
+  const group = getGroup(groupId);
   if (! group.admins || ! group.admins.includes(dismisser)) {
     throw new errors.NotAdminError();
   }
   deleteMembership(groupId, dismissee, timestamp);
 }
 
-function loadUser(id) {
-  return query`RETURN DOCUMENT(${usersColl}, ${id})`.toArray()[0];
-}
-
-function userScore(key) {
-  return query`
-    FOR u in ${usersColl}
-      FILTER u._key  == ${key}
-      RETURN u.score
-  `.toArray()[0];
+function getUser(id) {
+  if (! usersColl.exists(id)) {
+    throw new errors.UserNotFoundError(id);
+  }
+  return usersColl.document(id);
 }
+const checkUserExists = getUser;
 
 function createUser(key, timestamp) {
-  // already exists?
-  const user = loadUser(key);
-
-  if (!user) {
+  if (!usersColl.exists(key)) {
     return usersColl.insert({
-      score: 0,
       signingKeys: [urlSafeB64ToB64(key)],
       createdAt: timestamp,
       _key: key
     });
   } else {
-    return user;
+    return usersColl.document(key);
   }
 }
 
-function hasPrimaryGroup(key) {
-  const groupIds = usersInGroupsColl.byExample({
-    _from: 'users/' + key
-  }).toArray().map(ug => ug._to.replace('groups/', ''));
-  const groups = groupsColl.documents(groupIds).documents;
-  return groups.filter(group => group.type == 'primary').length > 0;
+function familyGroup(key) {
+  return userMemberships(key).map(
+    m => groupsColl.document(m.id)
+  ).find(g => g.type == 'family');
```

Comment:
Each person can belong to at most one family group, not including family groups where they are the head. This function correctly returns only one family group, but might return the one where they are the head.  It should filter out any groups where the user is the head.

Reply:
Done